### DISCUSSION TITLE
Zoom to area...

### DISCUSSION THREAD

**Discussion by keince** [pelicanmapping/osgearth#2163] [Support Q&A]
Does the osgEarth Earth manipulator have anything that supports a "zoom to area" functionality? In other words, if I define a rectangle in the current window, I'd like to zoom to that rectangle so it fills the entire window.

If the functionality doesn't exist, what would be the best approach for implementation?

Thanks in advance...

**Comment by gwaldron** [comment]
OOPS missed this when you first asked. 
Try the `ViewFitter` class, it does just that. The `LayersGUI` imgui panel uses it to zoom to an extent.
